metecho/api/models.py

```diff
@@ -541,6 +541,7 @@ class Task(
     branch_name = models.CharField(
         max_length=100, null=True, blank=True, validators=[validate_unicode_branch]
     )
+    org_config_name = StringField(blank=True)
```

Comment:
@wlonk Hm... I think it's correct for this to be an arbitrary `StringField`, but 1) I think it needs to be required, and 2) for each Task, the available options here should come directly from the org configs defined in `cumulusci.yml`. I'm not sure if that should be stored as a field on a Repository or a Project, and how often it might need to be refreshed if it changes? I'm also not sure what the default/fallback should be if `cumulusci.yml` doesn't have any `orgs.scratch` defined (just `dev`? or `dev` and `qa`?)... @davisagli could you weigh in here?

Cf. https://trello.com/c/Eta19Ib5/188-i-want-to-create-a-scratch-org-with-a-different-flow-or-org-type#comment-5ec2dac8e800a381d859b074 and https://trello.com/c/Eta19Ib5/188-i-want-to-create-a-scratch-org-with-a-different-flow-or-org-type#comment-5eb426f6584bca82a29093c8

In the example (https://trello.com/c/Eta19Ib5/188-i-want-to-create-a-scratch-org-with-a-different-flow-or-org-type#comment-5ec2dac8e800a381d859b074) the config was nested under `orgs.scratch` -- is that consistent?

Reply:
> for each Task, the available options here should come directly from the org configs defined in cumulusci.yml. I'm not sure if that should be stored as a field on a Repository or a Project, and how often it might need to be refreshed if it changes?

The options should come from the project_config.orgs__scratch dict (which merges orgs from a project's cumulusci.yml with ones inherited from the default cumulusci.yml). This means you'll have to construct the project config in the context of a particular commit since new orgs can be added by editing cumulusci.yml. It's probably best to cache this at the project level but provide some way to clear the cached list of orgs and/or edit it manually if necessary.

>  I'm also not sure what the default/fallback should be if cumulusci.yml doesn't have any orgs.scratch defined (just dev? or dev and qa?)

It will always have at least the orgs that are inherited from the default config (dev, qa, feature, beta, release)

> In the example the config was nested under orgs.scratch -- is that consistent?

Yes

Reply:
Implemented; @jgerigmeyer how do you wanna handle the refresh? Right now, there's a method on `Projects` to re-fetch these, but no endpoint to trigger that method. Let's discuss.

Reply:
@wlonk Yeah, I think from a user's perspective it'd be nice to add a subtle "refresh" btn next to the list of flow options when creating or editing a task. Would that work?

Reply:
Added `project-refresh-org-config-names` endpoint.

Reply:
@wlonk Forgot to push?